lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2063,6 +2063,20 @@ void TypeChecker::checkOwnershipAttr(VarDecl *var, OwnershipAttr *attr) {
 
     diagnose(var->getStartLoc(), D, (unsigned) ownershipKind, underlyingType);
     attr->setInvalid();
+    return;
+  } else if (var->getDeclContext()->getAsProtocolOrProtocolExtensionContext()) {
```

Comment:
This will also be true for protocol extensions. I don't think there should be any prohibition against defining weak or unowned properties in protocol extensions, and I don't see anything similar in the proposal. Please add a test that this case continues to work.

Reply:
We don't actually let you *put* stored properties in a protocol extension today (instance or type properties), so it won't make much of a difference, but Slava's right that it shouldn't get *this* diagnostic.

Reply:
Do we not allowed computed weak/unowned properties? I guess that doesn't make much sense, but I wasn't aware of the restriction.

Reply:
Huh, it looks like we do. In that case, I'm not sure whether this is a good idea or not, but that's an issue for the review, not for the implementation.

Reply:
Perhaps we should broaden the SE proposal to ban computed properties with ownership qualifiers also. It doesn't really make any sense to declare a weak or unowned computed property in my opinion, since this is a property of the implementation of the underlying storage and not of the contract of the property itself. The latter is already captured wholly by giving the property an optional or IUO type, in my opinion.

Yay for implementing proposals during review and teasing out odd corner cases!